nuxeo-drive-client/nxdrive/tests/test_engine_offline.py

```diff
@@ -82,17 +96,13 @@ def test_with_mock_server_offline(self):
 
         # Stop mocking and call original method
         EngineOffLineTestCase.server_online = True
-        # Wait for queue_manager to process for the folder/file
-        sleep(60)
 
-        # pair_state should be 'synchronized' for the folder/file
-        test_folder = self.get_dao_state_from_engine_1('/FolderA')
-        self.assertEqual(test_folder.pair_state, 'synchronized')
-        test_file = self.get_dao_state_from_engine_1('/FolderA/TestFile.txt')
-        self.assertEqual(test_file.pair_state, 'synchronized')
+        # Wait for files to upload
+        states = self.wait_for_upload(items=['/FolderA', '/FolderA/TestFile.txt'])
+        for state in states:
```

Comment:
wait_for_upload could just return a boolean ('False' if timeout), so no need to loop again thru the list here